lib/redis_mutex.ex

```diff
@@ -1,90 +1,135 @@
 defmodule RedisMutex do
   @moduledoc """
-  An Elixir library for using Redis locks
+  An Elixir library for using Redis locks.
+  """
 
-  ## Setup
+  @type connection_options :: [
+          host: String.t(),
+          port: non_neg_integer(),
+          database: String.t() | non_neg_integer(),
+          username: String.t(),
+          password: Redix.password(),
+          timeout: timeout(),
+          sync_connect: boolean(),
+          exit_on_disconnection: boolean(),
+          backoff_initial: non_neg_integer(),
+          backoff_max: timeout(),
+          ssl: boolean(),
+          socket_opts: list(term()),
+          hibernate_after: non_neg_integer(),
+          spawn_opt: keyword(),
```

Comment:
is there a type we can point to?

Reply:
This is taken directly from Redix's documentation

> `:spawn_opt `([`keyword/0`](https://hexdocs.pm/elixir/typespecs.html#built-in-types)) - if present, its value is passed as options to the Redix connection process as in [`Process.spawn/4`](https://hexdocs.pm/elixir/Process.html#spawn/4). See [`:gen_statem.start_opt/0`](https://www.erlang.org/doc/man/gen_statem.html#type-start_opt). Not present by default.

https://hexdocs.pm/redix/Redix.html#start_link/1-options